File: flutter/lib/pantallas/seccion3_seleccion.dart
import 'package:flutter/material.dart';

import '../estado.dart';
import '../secciones.dart';
import '../widgets/comunes.dart';

const _lenguajes = ['Kotlin', 'Dart', 'Java', 'Swift', 'C#', 'JavaScript'];
const _intereses = ['Música', 'Deportes', 'Tecnología', 'Viajes', 'Cine', 'Arte'];

class Seccion3Seleccion extends StatelessWidget {
  const Seccion3Seleccion({super.key});

  @override
  Widget build(BuildContext context) {
    return const PantallaSeccion(
      seccion: Seccion.seleccion,
      elementos: [
        ElementoCatalogo(
          nombre: 'Casilla de verificación',
          componente: 'Checkbox(tristate: true) / CheckboxListTile',
          descripcion: 'Permite marcar varias opciones de forma independiente. La casilla '
              'principal muestra un estado indeterminado cuando solo algunas opciones están marcadas.',
          demo: _Casillas(),
        ),
        ElementoCatalogo(
          nombre: 'Botones de opción (conectado con la Sección 5)',
          componente: 'RadioGroup + RadioListTile',
          descripcion: 'Permiten elegir una sola opción de un grupo. La opción elegida aquí '
              'define el tamaño de los textos de ejemplo que se muestran en la Sección 5.',
          demo: _BotonesOpcion(),
        ),
        ElementoCatalogo(
          nombre: 'Interruptor',
          componente: 'Switch / SwitchListTile',
          descripcion: 'Activa o desactiva una opción de forma inmediata, sin necesidad de '
              'confirmar. Es ideal para ajustes que se aplican al momento.',
          demo: _Interruptores(),
        ),
        ElementoCatalogo(
          nombre: 'Deslizador de valor único y de rango',
          componente: 'Slider / RangeSlider',
          descripcion: 'Permiten elegir un valor numérico arrastrando un control sobre una '
              'pista. El de rango tiene dos controles para definir un mínimo y un máximo.',
          demo: _Deslizadores(),
        ),
        ElementoCatalogo(
          nombre: 'Lista desplegable',
          componente: 'DropdownMenu<String>',
          descripcion: 'Muestra la opción seleccionada y, al tocarla, despliega un menú con '
              'todas las alternativas. Ahorra espacio cuando hay muchas opciones.',
          demo: _ListaDesplegable(),
        ),
        ElementoCatalogo(
          nombre: 'Selector de fecha y de hora',
          componente: 'showDatePicker / showTimePicker',
          descripcion: 'Diálogos que muestran un calendario o un reloj para elegir una fecha '
              'o una hora sin tener que escribirla, evitando errores de formato.',
          demo: _SelectoresFechaHora(),
        ),
        ElementoCatalogo(
          nombre: 'Chips de filtro',
          componente: 'FilterChip',
          descripcion: 'Etiquetas compactas que se activan o desactivan para filtrar contenido. '
              'Se pueden seleccionar varias a la vez y muestran una marca cuando están activas.',
          demo: _ChipsFiltro(),
        ),
      ],
    );
  }
}

class _Casillas extends StatefulWidget {
  const _Casillas();

  @override
  State<_Casillas> createState() => _CasillasState();
}

class _CasillasState extends State<_Casillas> {
  static const _ingredientes = ['Queso', 'Jamón', 'Champiñones'];
  final _marcados = [true, false, false];

  /// true = todas, false = ninguna, null = indeterminado.
  bool? get _general {
    if (_marcados.every((m) => m)) return true;
    if (_marcados.every((m) => !m)) return false;
    return null;
  }

  @override
  Widget build(BuildContext context) {
    final general = _general;
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        CheckboxListTile(
          contentPadding: EdgeInsets.zero,
          controlAffinity: ListTileControlAffinity.leading,
          tristate: true,
          value: general,
          title: const Text('Todos los ingredientes'),
          onChanged: (_) => setState(() {
            final nuevo = general != true;
            for (var i = 0; i < _marcados.length; i++) {
              _marcados[i] = nuevo;
            }
          }),
        ),
        for (var i = 0; i < _ingredientes.length; i++)
          CheckboxListTile(
            contentPadding: const EdgeInsets.only(left: 24),
            controlAffinity: ListTileControlAffinity.leading,
            dense: true,
            value: _marcados[i],
            title: Text(_ingredientes[i]),
            onChanged: (v) => setState(() => _marcados[i] = v ?? false),
          ),
        TextoResultado(
          'Estado de la casilla principal: ${general == null ? 'indeterminado' : general ? 'marcado' : 'desmarcado'}',
        ),
      ],
    );
  }
}

class _BotonesOpcion extends StatelessWidget {
  const _BotonesOpcion();

  @override
  Widget build(BuildContext context) {
    final estado = EstadoScope.of(context);
    final base = Theme.of(context).textTheme.bodyLarge!;
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        RadioGroup<TamanoTexto>(
          groupValue: estado.tamanoTexto,
          onChanged: (v) {
            if (v != null) estado.tamanoTexto = v;
          },
          child: Column(
            children: [
              for (final t in TamanoTexto.values)
                RadioListTile<TamanoTexto>(
                  contentPadding: EdgeInsets.zero,
                  dense: true,
                  value: t,
                  title: Text(t.etiqueta),
                ),
            ],
          ),
        ),
        Text('Vista previa del texto', style: base.copyWith(fontSize: base.fontSize! * estado.tamanoTexto.escala)),
        espacio,
        TextoResultado('Tamaño elegido: ${estado.tamanoTexto.etiqueta}. Revisa la Sección 5.'),
      ],
    );
  }
}

class _Interruptores extends StatefulWidget {
  const _Interruptores();

  @override
  State<_Interruptores> createState() => _InterruptoresState();
}

class _InterruptoresState extends State<_Interruptores> {
  bool _avion = false;
  bool _wifi = true;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          children: [
            const Icon(Icons.airplanemode_active),
            const SizedBox(width: 12),
            const Expanded(child: Text('Modo avión')),
            Switch(
              value: _avion,
              thumbIcon: WidgetStateProperty.resolveWith(
                (estados) => Icon(estados.contains(WidgetState.selected) ? Icons.check : Icons.close),
              ),
              onChanged: (v) => setState(() {
                _avion = v;
                if (v) _wifi = false;
              }),
            ),
          ],
        ),
        SwitchListTile(
          contentPadding: EdgeInsets.zero,
          title: const Text('Wi-Fi'),
          value: _wifi,
          onChanged: _avion ? null : (v) => setState(() => _wifi = v),
        ),
        TextoResultado(
          _avion ? 'Modo avión activado: conexiones desactivadas' : 'Wi-Fi ${_wifi ? 'encendido' : 'apagado'}',
        ),
      ],
    );
  }
}

class _Deslizadores extends StatefulWidget {
  const _Deslizadores();

  @override
  State<_Deslizadores> createState() => _DeslizadoresState();
}

class _DeslizadoresState extends State<_Deslizadores> {
  double _volumen = 40;
  RangeValues _precio = const RangeValues(200, 800);

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text('Volumen: ${_volumen.round()} %'),
        Slider(
          value: _volumen,
          max: 100,
          label: '${_volumen.round()} %',
          onChanged: (v) => setState(() => _volumen = v),
        ),
        Text('Precio: \$${_precio.start.round()} – \$${_precio.end.round()} MXN'),
        RangeSlider(
          values: _precio,
          max: 1000,
          divisions: 20,
          labels: RangeLabels('\$${_precio.start.round()}', '\$${_precio.end.round()}'),
          onChanged: (v) => setState(() => _precio = v),
        ),
      ],
    );
  }
}

class _ListaDesplegable extends StatefulWidget {
  const _ListaDesplegable();

  @override
  State<_ListaDesplegable> createState() => _ListaDesplegableState();
}

class _ListaDesplegableState extends State<_ListaDesplegable> {
  String _lenguaje = _lenguajes.first;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        DropdownMenu<String>(
          initialSelection: _lenguaje,
          label: const Text('Lenguaje favorito'),
          expandedInsets: EdgeInsets.zero,
          requestFocusOnTap: false,
          onSelected: (v) {
            if (v != null) setState(() => _lenguaje = v);
          },
          dropdownMenuEntries: [
            for (final l in _lenguajes) DropdownMenuEntry(value: l, label: l),
          ],
        ),
        espacio,
        TextoResultado('Elegiste: $_lenguaje'),
      ],
    );
  }
}

class _SelectoresFechaHora extends StatefulWidget {
  const _SelectoresFechaHora();

  @override
  State<_SelectoresFechaHora> createState() => _SelectoresFechaHoraState();
}

class _SelectoresFechaHoraState extends State<_SelectoresFechaHora> {
  DateTime? _fecha;
  TimeOfDay? _hora;

  Future<void> _elegirFecha() async {
    final hoy = DateTime.now();
    final elegida = await showDatePicker(
      context: context,
      initialDate: _fecha ?? hoy,
      firstDate: DateTime(hoy.year - 50),
      lastDate: DateTime(hoy.year + 10),
    );
    if (elegida != null) setState(() => _fecha = elegida);
  }

  Future<void> _elegirHora() async {
    final elegida = await showTimePicker(
      context: context,
      initialTime: _hora ?? TimeOfDay.now(),
      helpText: 'Selecciona la hora',
      builder: (context, hijo) =>
          MediaQuery(data: MediaQuery.of(context).copyWith(alwaysUse24HourFormat: true), child: hijo!),
    );
    if (elegida != null) setState(() => _hora = elegida);
  }

  @override
  Widget build(BuildContext context) {
    final loc = MaterialLocalizations.of(context);
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Wrap(
          spacing: 8,
          runSpacing: 8,
          children: [
            OutlinedButton.icon(
              onPressed: _elegirFecha,
              icon: const Icon(Icons.calendar_month),
              label: const Text('Elegir fecha'),
            ),
            OutlinedButton.icon(
              onPressed: _elegirHora,
              icon: const Icon(Icons.schedule),
              label: const Text('Elegir hora'),
            ),
          ],
        ),
        espacio,
        TextoResultado('Fecha: ${_fecha == null ? 'sin elegir' : loc.formatFullDate(_fecha!)}'),
        TextoResultado('Hora: ${_hora == null ? 'sin elegir' : loc.formatTimeOfDay(_hora!, alwaysUse24HourFormat: true)}'),
      ],
    );
  }
}

class _ChipsFiltro extends StatefulWidget {
  const _ChipsFiltro();

  @override
  State<_ChipsFiltro> createState() => _ChipsFiltroState();
}

class _ChipsFiltroState extends State<_ChipsFiltro> {
  final _seleccion = <String>{'Tecnología'};

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Wrap(
          spacing: 8,
          runSpacing: 4,
          children: [
            for (final interes in _intereses)
              FilterChip(
                label: Text(interes),
                selected: _seleccion.contains(interes),
                onSelected: (v) => setState(() => v ? _seleccion.add(interes) : _seleccion.remove(interes)),
              ),
          ],
        ),
        espacio,
        TextoResultado(
          _seleccion.isEmpty
              ? 'Sin filtros activos'
              : 'Filtros activos (${_seleccion.length}): ${_intereses.where(_seleccion.contains).join(', ')}',
        ),
      ],
    );
  }
}

File: android-compose/app/src/main/java/com/javiergamez/catalogoui/compose/secciones/Seccion3Seleccion.kt
@file:OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)

package com.javiergamez.catalogoui.compose.secciones

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.selection.selectable
import androidx.compose.foundation.selection.selectableGroup
import androidx.compose.foundation.selection.toggleable
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AirplanemodeActive
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Checkbox
import androidx.compose.material3.DatePicker
import androidx.compose.material3.DatePickerDialog
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuAnchorType
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.RadioButton
import androidx.compose.material3.RangeSlider
import androidx.compose.material3.Slider
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TimePicker
import androidx.compose.material3.TriStateCheckbox
import androidx.compose.material3.rememberDatePickerState
import androidx.compose.material3.rememberTimePickerState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableFloatStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.state.ToggleableState
import androidx.compose.ui.unit.dp
import com.javiergamez.catalogoui.compose.componentes.ElementoCatalogo
import com.javiergamez.catalogoui.compose.componentes.PantallaSeccion
import com.javiergamez.catalogoui.compose.componentes.TextoResultado
import com.javiergamez.catalogoui.compose.datos.CatalogoViewModel
import com.javiergamez.catalogoui.compose.datos.TamanoTexto
import com.javiergamez.catalogoui.compose.navegacion.Seccion
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone
import kotlin.math.roundToInt

private val LENGUAJES = listOf("Kotlin", "Dart", "Java", "Swift", "C#", "JavaScript")
private val INTERESES = listOf("Música", "Deportes", "Tecnología", "Viajes", "Cine", "Arte")

@Composable
fun Seccion3Seleccion(vm: CatalogoViewModel) {
    PantallaSeccion(Seccion.SELECCION) {
        item {
            ElementoCatalogo(
                nombre = "Casilla de verificación",
                componente = "Checkbox / TriStateCheckbox",
                descripcion = "Permite marcar varias opciones de forma independiente. La casilla " +
                    "principal muestra un estado indeterminado cuando solo algunas opciones están marcadas.",
            ) {
                val ingredientes = listOf("Queso", "Jamón", "Champiñones")
                var marcados by rememberSaveable { mutableStateOf(listOf(true, false, false)) }
                val estadoGeneral = when {
                    marcados.all { it } -> ToggleableState.On
                    marcados.none { it } -> ToggleableState.Off
                    else -> ToggleableState.Indeterminate
                }
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.toggleable(
                        value = estadoGeneral == ToggleableState.On,
                        role = Role.Checkbox,
                        onValueChange = {
                            val nuevo = estadoGeneral != ToggleableState.On
                            marcados = marcados.map { nuevo }
                        },
                    ),
                ) {
                    TriStateCheckbox(state = estadoGeneral, onClick = null)
                    Text("Todos los ingredientes", style = MaterialTheme.typography.titleSmall)
                }
                Column(Modifier.padding(start = 24.dp)) {
                    ingredientes.forEachIndexed { i, nombre ->
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            modifier = Modifier.toggleable(
                                value = marcados[i],
                                role = Role.Checkbox,
                                onValueChange = { v -> marcados = marcados.toMutableList().also { it[i] = v } },
                            ),
                        ) {
                            Checkbox(checked = marcados[i], onCheckedChange = null)
                            Text(nombre)
                        }
                    }
                }
                val estadoTexto = when (estadoGeneral) {
                    ToggleableState.On -> "marcado"
                    ToggleableState.Off -> "desmarcado"
                    ToggleableState.Indeterminate -> "indeterminado"
                }
                TextoResultado("Estado de la casilla principal: $estadoTexto")
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Botones de opción (conectado con la Sección 5)",
                componente = "RadioButton + Modifier.selectableGroup",
                descripcion = "Permiten elegir una sola opción de un grupo. La opción elegida aquí " +
                    "define el tamaño de los textos de ejemplo que se muestran en la Sección 5.",
            ) {
                Column(Modifier.selectableGroup()) {
                    TamanoTexto.entries.forEach { tamano ->
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            modifier = Modifier
                                .fillMaxWidth()
                                .selectable(
                                    selected = vm.tamanoTexto == tamano,
                                    onClick = { vm.tamanoTexto = tamano },
                                    role = Role.RadioButton,
                                ),
                        ) {
                            RadioButton(selected = vm.tamanoTexto == tamano, onClick = null)
                            Text(tamano.etiqueta, modifier = Modifier.padding(start = 8.dp))
                        }
                    }
                }
                Text(
                    "Vista previa del texto",
                    fontSize = MaterialTheme.typography.bodyLarge.fontSize * vm.tamanoTexto.escala,
                )
                TextoResultado("Tamaño elegido: ${vm.tamanoTexto.etiqueta}. Revisa la Sección 5.")
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Interruptor",
                componente = "Switch",
                descripcion = "Activa o desactiva una opción de forma inmediata, sin necesidad de " +
                    "confirmar. Es ideal para ajustes que se aplican al momento.",
            ) {
                var avion by rememberSaveable { mutableStateOf(false) }
                var wifi by rememberSaveable { mutableStateOf(true) }
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(Icons.Filled.AirplanemodeActive, contentDescription = null)
                    Text("Modo avión", modifier = Modifier
                        .weight(1f)
                        .padding(start = 12.dp))
                    Switch(
                        checked = avion,
                        onCheckedChange = {
                            avion = it
                            if (it) wifi = false
                        },
                        thumbContent = {
                            Icon(
                                if (avion) Icons.Filled.Check else Icons.Filled.Close,
                                contentDescription = null,
                                modifier = Modifier.size(SwitchDefaults.IconSize),
                            )
                        },
                    )
                }
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text("Wi-Fi", modifier = Modifier.weight(1f))
                    Switch(checked = wifi, onCheckedChange = { wifi = it }, enabled = !avion)
                }
                TextoResultado(
                    if (avion) "Modo avión activado: conexiones desactivadas"
                    else "Wi-Fi ${if (wifi) "encendido" else "apagado"}",
                )
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Deslizador de valor único y de rango",
                componente = "Slider / RangeSlider",
                descripcion = "Permiten elegir un valor numérico arrastrando un control sobre una " +
                    "pista. El de rango tiene dos controles para definir un mínimo y un máximo.",
            ) {
                var volumen by rememberSaveable { mutableFloatStateOf(40f) }
                var desde by rememberSaveable { mutableFloatStateOf(200f) }
                var hasta by rememberSaveable { mutableFloatStateOf(800f) }
                Text("Volumen: ${volumen.roundToInt()} %")
                Slider(value = volumen, onValueChange = { volumen = it }, valueRange = 0f..100f)
                Text("Precio: $${desde.roundToInt()} – $${hasta.roundToInt()} MXN")
                RangeSlider(
                    value = desde..hasta,
                    onValueChange = {
                        desde = it.start
                        hasta = it.endInclusive
                    },
                    valueRange = 0f..1000f,
                    steps = 19,
                )
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Lista desplegable",
                componente = "ExposedDropdownMenuBox (solo lectura)",
                descripcion = "Muestra la opción seleccionada y, al tocarla, despliega un menú con " +
                    "todas las alternativas. Ahorra espacio cuando hay muchas opciones.",
            ) {
                var expandido by rememberSaveable { mutableStateOf(false) }
                var lenguaje by rememberSaveable { mutableStateOf(LENGUAJES.first()) }
                ExposedDropdownMenuBox(expanded = expandido, onExpandedChange = { expandido = it }) {
                    OutlinedTextField(
                        value = lenguaje,
                        onValueChange = {},
                        readOnly = true,
                        label = { Text("Lenguaje favorito") },
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expandido) },
                        modifier = Modifier
                            .menuAnchor(ExposedDropdownMenuAnchorType.PrimaryNotEditable)
                            .fillMaxWidth(),
                    )
                    ExposedDropdownMenu(expanded = expandido, onDismissRequest = { expandido = false }) {
                        LENGUAJES.forEach { opcion ->
                            DropdownMenuItem(
                                text = { Text(opcion) },
                                onClick = {
                                    lenguaje = opcion
                                    expandido = false
                                },
                            )
                        }
                    }
                }
                TextoResultado("Elegiste: $lenguaje")
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Selector de fecha y de hora",
                componente = "DatePickerDialog / TimePicker",
                descripcion = "Diálogos que muestran un calendario o un reloj para elegir una fecha " +
                    "o una hora sin tener que escribirla, evitando errores de formato.",
            ) {
                var fecha by rememberSaveable { mutableStateOf<Long?>(null) }
                var hora by rememberSaveable { mutableStateOf<String?>(null) }
                var mostrarFecha by rememberSaveable { mutableStateOf(false) }
                var mostrarHora by rememberSaveable { mutableStateOf(false) }
                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    OutlinedButton(onClick = { mostrarFecha = true }) {
                        Icon(Icons.Filled.CalendarMonth, contentDescription = null, modifier = Modifier.size(18.dp))
                        Text("  Elegir fecha")
                    }
                    OutlinedButton(onClick = { mostrarHora = true }) {
                        Icon(Icons.Filled.Schedule, contentDescription = null, modifier = Modifier.size(18.dp))
                        Text("  Elegir hora")
                    }
                }
                val formato = remember { SimpleDateFormat("d 'de' MMMM 'de' yyyy", Locale.forLanguageTag("es-MX")).apply { timeZone = TimeZone.getTimeZone("UTC") } }
                TextoResultado("Fecha: ${fecha?.let { formato.format(Date(it)) } ?: "sin elegir"}")
                TextoResultado("Hora: ${hora ?: "sin elegir"}")

                if (mostrarFecha) {
                    val estado = rememberDatePickerState(initialSelectedDateMillis = fecha)
                    DatePickerDialog(
                        onDismissRequest = { mostrarFecha = false },
                        confirmButton = {
                            TextButton(onClick = {
                                fecha = estado.selectedDateMillis
                                mostrarFecha = false
                            }) { Text("Aceptar") }
                        },
                        dismissButton = {
                            TextButton(onClick = { mostrarFecha = false }) { Text("Cancelar") }
                        },
                    ) {
                        DatePicker(state = estado)
                    }
                }
                if (mostrarHora) {
                    val estado = rememberTimePickerState(is24Hour = true)
                    AlertDialog(
                        onDismissRequest = { mostrarHora = false },
                        title = { Text("Selecciona la hora") },
                        text = { TimePicker(state = estado) },
                        confirmButton = {
                            TextButton(onClick = {
                                hora = "%02d:%02d".format(estado.hour, estado.minute)
                                mostrarHora = false
                            }) { Text("Aceptar") }
                        },
                        dismissButton = {
                            TextButton(onClick = { mostrarHora = false }) { Text("Cancelar") }
                        },
                    )
                }
            }
        }

        item {
            ElementoCatalogo(
                nombre = "Chips de filtro",
                componente = "FilterChip",
                descripcion = "Etiquetas compactas que se activan o desactivan para filtrar contenido. " +
                    "Se pueden seleccionar varias a la vez y muestran una marca cuando están activas.",
            ) {
                var seleccion by rememberSaveable { mutableStateOf(listOf("Tecnología")) }
                FlowRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    INTERESES.forEach { interes ->
                        val activo = interes in seleccion
                        FilterChip(
                            selected = activo,
                            onClick = { seleccion = if (activo) seleccion - interes else seleccion + interes },
                            label = { Text(interes) },
                            leadingIcon = if (activo) {
                                { Icon(Icons.Filled.Check, contentDescription = null, modifier = Modifier.size(FilterChipDefaults.IconSize)) }
                            } else {
                                null
                            },
                        )
                    }
                }
                TextoResultado(
                    if (seleccion.isEmpty()) "Sin filtros activos"
                    else "Filtros activos (${seleccion.size}): ${seleccion.joinToString()}",
                )
            }
        }
    }
}
